src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -97,101 +108,116 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
 
             var message = formatter(state, exception);
 
-            if (!string.IsNullOrEmpty(message))
-            {
-                WriteMessage(logLevel, Name, eventId.Id, message);
-            }
-
-            if (exception != null)
+            if (!string.IsNullOrEmpty(message) || exception != null)
             {
-                WriteException(logLevel, Name, eventId.Id, exception);
+                WriteMessage(logLevel, Name, eventId.Id, message, exception);
             }
         }
 
-        public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message)
+        public virtual void WriteMessage(LogLevel logLevel, string logName, int eventId, string message, Exception exception)
         {
-            // check if the message has any new line characters in it and provide the padding if necessary
-            message = ReplaceMessageNewLinesWithPadding(message);
-            var logLevelColors = GetLogLevelConsoleColors(logLevel);
-            var loglevelString = GetLogLevelString(logLevel);
+            var fullMessageBuilder = new StringBuilder();
+
+            ConsoleColors logLevelColors = DefaultConsoleColors;
+            string loglevelString = string.Empty;
+            bool loglevelUsesDefaultColors = true;
 
             // Example:
             // INFO: ConsoleApp.Program[10]
             //       Request received
-
-            lock (_lock)
+            if(!string.IsNullOrEmpty(message))
             {
-                // log level string
-                WriteWithColor(
-                    logLevelColors.Foreground,
-                    logLevelColors.Background,
-                    loglevelString,
-                    newLine: false);
+                logLevelColors = GetLogLevelConsoleColors(logLevel);
+                loglevelString = GetLogLevelString(logLevel);
+                loglevelUsesDefaultColors = DefaultConsoleColors.Equals(logLevelColors);
 
-                // category and event id
-                // use default colors
-                WriteWithColor(
-                    DefaultConsoleColor,
-                    DefaultConsoleColor,
-                    _loglevelPadding + logName + $"[{eventId}]",
-                    newLine: true);
+                if (loglevelUsesDefaultColors)
```

Comment:
I understand it now, however it seems you don't need an if and can just assume the logLevel doesn't use default colors?